x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -879,8 +892,13 @@ String toQuotedJsonArray(Object[] values) {
     }
 
     private static String effectiveRealmName(Authentication authentication) {
-        return authentication.getLookedUpBy() != null ? authentication.getLookedUpBy().getName()
-                : authentication.getAuthenticatedBy().getName();
+        if (authentication.getLookedUpBy() != null) {
+            return authentication.getLookedUpBy().getName();
+        } else if (authentication.getAuthenticationType() == Authentication.AuthenticationType.API_KEY) {
+            return (String) authentication.getMetadata().get(ApiKeyService.API_KEY_CREATOR_REALM_NAME);
+        } else {
+            return authentication.getAuthenticatedBy().getName();
+        }
```

Comment:
This method is now a duplicate of `ApiKeyService#getCreatorRealmName`. It is probably better to consolidate them.

Reply:
Good suggestion, thank you!